### PR TITLE
Add support for packet filtering of incoming and outgoing packets

### PR DESCRIPTION
**Pull request by dpino**
When parameter `filter` is set in lwAftr conf file. Example:

```
filter="ip or ip6"
```

where `filter` is a pcap lang expression.

### CODE REVIEW

File: src/program/snabb_lwaftr/setup.lua [hunk 1]
```diff
@@ -22,10 +23,26 @@ function lwaftr_app(c, conf)
    config.app(c, "fragmenterv6", ipv6_apps.Fragmenter,
               { mtu=conf.ipv6_mtu })
 
-   config.link(c, "reassemblerv4.output -> lwaftr.v4")
```
Comment (kbara) [inline src/program/snabb_lwaftr/setup.lua]: This is getting horribly non-compositional. I've made changes to the same code for NDP resolution. Any thoughts on keeping the complexity under control?

**Comment by kbara** [comment]
After I've heard from you about compositionality, LGTM.

File: src/apps/lwaftr/conf.lua [hunk 2]
```diff
@@ -48,7 +48,8 @@ local lwaftr_conf_spec = {
       policy_icmpv6_outgoing=Parser.enum_parser(policies),
       v4_vlan_tag=Parser.parse_vlan_tag,
       v6_vlan_tag=Parser.parse_vlan_tag,
-      vlan_tagging=Parser.parse_boolean
+      vlan_tagging=Parser.parse_boolean,
+      filter=Parser.parse_string,
```
Comment (wingo) [inline src/apps/lwaftr/conf.lua]: I think we need to be able to specify filters per-interface.  e.g. ipv4_ingress_filter, ipv4_egress_filter, ipv6_ingress_filter, ipv6_egress_filter

**Comment by dpino** [comment]
@kbara To tackle this problem, I think we can define an array of preprocessing apps and another array of postprocessing apps. A helper function links the preprocessing apps together (same for postprocessing apps). Adding a NDP app would be adding a new postprocessing app to the array. WDYT?

**Comment by kbara** [comment]
It's not just post-processing, it's bi-directional.

**Comment by dpino** [comment]
@kbara OK, I just saw your PR. Does this solution works for your or you prefer something else?

**Comment by dpino** [comment]
Updated PR. With these changes NDP configuration could be done as follows:

```
if conf.use_ndp_resolution then
   config.app(c, "ndp_eth_next_hop", ipv6_apps.NDP_eth_next_hop,
              { src_ipv6 = conf.aftr_ipv6_ip, src_eth = conf.aftr_mac_b4_side,
                dst_ipv6 = conf.next_hop_ipv6_addr  })
   append(preprocessing_apps_v6, { name = "ndp_eth_next_hop", input = "south", output = "north" })
   append(postprocessing_apps_v6, { name = "ndp_eth_next_hop", input = "south", output = "north" })
end
```

**Comment by kbara** [comment]
LGTM, thanks. @wingo , wdyt?

**Comment by wingo** [comment]
LGTM.  Thank you @dpino!

**Pull request merged**